Extension/UqlFunctionCaller.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Netdudes\DataSourceryBundle\Extension;
+
+use Netdudes\DataSourceryBundle\Extension\Exception\FunctionNotFoundException;
+
+class UqlFunctionCaller
+{
+    /**
+     * @var UqlExtensionContainer
+     */
+    private $uqlExtensionContainer;
+
+    /**
+     * @param UqlExtensionContainer $uqlExtensionContainer
+     */
+    public function __construct(UqlExtensionContainer $uqlExtensionContainer)
+    {
+        $this->uqlExtensionContainer = $uqlExtensionContainer;
+    }
+
+    /**
+     * @param string  $name
+     * @param array   $arguments
+     * @param Context $context
+     *
+     * @return mixed
+     * @throws Exception\FunctionNotFoundException
+     */
+    public function callFunction($name, $arguments, Context $context)
```

Comment:
now you *always* have to pass a contxt if you want to call a function? is there always a context?